pkg/querier/worker/frontend_processor_test.go

```diff
@@ -0,0 +1,69 @@
+package worker
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+	"google.golang.org/grpc"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/test"
+)
+
+func TestRecvFailDoesntCancelProcess(t *testing.T) {
```

Comment:
I'm missing a thing. How is it asserted that doesn't cancel process?

Reply:
The point of this test is to verify that frontend processor's `processQueriesOnSingleStream` keeps running even though it cannot receive any queries from the frontend.